Source/FieldSolver/WarpXPushFieldsHybrid.cpp

```diff
@@ -0,0 +1,258 @@
+/* Copyright 2023 The WarpX Community
+ *
+ * This file is part of WarpX.
+ *
+ * Authors: Roelof Groenewald (TAE Technologies)
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+#include "Evolve/WarpXDtType.H"
+#include "FieldSolver/FiniteDifferenceSolver/FiniteDifferenceSolver.H"
+#include "FieldSolver/FiniteDifferenceSolver/HybridModel/HybridModel.H"
+#include "Utils/TextMsg.H"
+#include "Utils/WarpXProfilerWrapper.H"
+#include "WarpX.H"
+
+using namespace amrex;
+
+void WarpX::HybridEvolveFields ()
+{
+    // get requested number of substeps to use
+    int sub_steps = m_hybrid_model->m_substeps / 2;
+
+    // During the particle push and deposition (which already happened) the
+    // charge density and current density were updated. So that at this time we
+    // have rho^{n} in the 0'th index and rho{n+1} in the 1'st index of
+    // `rho_fp`, J_i^{n+1/2} in `current_fp` and J_i^{n-1/2} in
+    // `current_fp_temp`.
+
+    // TODO: insert Runge-Kutta integration logic for B update instead
+    // of the substep update used here - can test with small timestep using
+    // this simpler implementation
+
+    // Note: E^{n} is recalculated with the accurate J_i^{n} since at the end
+    // of the last step we had to "guess" J_i^{n}. It also needs to be
+    // recalculated to include the resistivity before evolving B.
+
+    // Firstly J_i^{n} is calculated as the average of J_i^{n-1/2} and J_i^{n+1/2}
+    for (int lev = 0; lev <= finest_level; ++lev)
+    {
+        // Loop through the grids, and over the tiles within each grid
+#ifdef AMREX_USE_OMP
+#pragma omp parallel if (amrex::Gpu::notInLaunchRegion())
+#endif
+        for ( MFIter mfi(*current_fp[lev][0], TilingIfNotGPU()); mfi.isValid(); ++mfi ) {
```

Comment:
I'm thinking this should be doable with simpler multifab operations. All that is happening here is `current_fp_temp = 0.5 * (current_fp_temp + current_fp)`.